### PR TITLE
Refactor remote namespaces creation

### PR DESCRIPTION
**Pull request by giorio94**
# Description

This PR refactors the creation of remote namespaces, moving the enforcement logic to the remote cluster in order to prevent issues in case of connectivity issues, as well as removing the need for the integration with capsule, which is dropped.

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [x] Manual testing
- [x] Unit testing (new + existing)
- [x] E2E testing (existing)

### CODE REVIEW

File: internal/crdReplicator/reflection/reflector.go [hunk 1]
```diff
@@ -179,16 +179,18 @@ func (r *Reflector) stopForResource(gvr schema.GroupVersionResource) error {
 
 	klog.Infof("[%v] Stopping reflection of %v", r.remoteClusterID, gvr)
 
-	// Check if any reflected object is still present in the remote cluster
-	objects, err := rs.remote.List(labels.Everything())
-	if err != nil {
-		klog.Errorf("[%v] Failed to stop reflection of %v: %v", r.remoteClusterID, gvr, err)
-		return err
-	}
+	// Check if any object is still present in the local or in the remote cluster
```
Comment (alacuku) [inline internal/crdReplicator/reflection/reflector.go:182]: Being the local objects owned by components other than the CRDReplicator shouldn't be them the ones that handle lifecycle for those resources?

**Review by alacuku — APPROVED** [review]

File: pkg/liqo-controller-manager/foreign-cluster-operator/permission.go [hunk 2]
```diff
@@ -47,12 +47,6 @@ func (r *ForeignClusterReconciler) ensurePermission(ctx context.Context, foreign
 			klog.Error(err)
 			return err
 		}
-		if r.OwnerReferencesPermissionEnforcement {
-			if err = r.NamespaceManager.UnbindIncomingClusterWideRole(ctx, remoteCluster); err != nil {
-				klog.Error(err)
-				return err
-			}
-		}
```
Comment (aleoli) [inline pkg/liqo-controller-manager/foreign-cluster-operator/permission.go:55]: Is this change necessary? (the same for the binding)

They are managing the cluster role binding that I think is not related to the tenant resource

Comment (giorio94) [reply pkg/liqo-controller-manager/foreign-cluster-operator/permission.go:55]: It seems to me the ClusterRole is related to the tenant finalizer only, if I haven't misinterpreted something. 
https://github.com/liqotech/liqo/blob/fadcbcd767b7d60959d3a53ee311d1cf91bdd4a2/pkg/tenantNamespace/clusterwide.go#L49-L56

Comment (giorio94) [reply pkg/liqo-controller-manager/foreign-cluster-operator/permission.go:55]: BTW, do the E2E tests run with the strict permissions mode?

Comment (aleoli) [reply pkg/liqo-controller-manager/foreign-cluster-operator/permission.go:55]: ok, yes you are right 👍

no, the e2e are running with the default mode (if I remember correctly). We should change it

**Review by aleoli — APPROVED** [review]
